rusk/tests/services/transactions.rs

```diff
@@ -0,0 +1,312 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::common::*;
+use canonical::{Canon, Source};
+use dusk_bls12_381_sign::PublicKey;
+use dusk_pki::{Ownable, SecretSpendKey, ViewKey};
+use dusk_schnorr::{PublicKeyPair, Signature};
+use parking_lot::Mutex;
+use rusk::services::prover::ExecuteProverRequest;
+use rusk::services::rusk_proto::prover_client::ProverClient;
+use rusk::services::rusk_proto::state_client::StateClient;
+use rusk::services::rusk_proto::Transaction as TransactionProto;
+use rusk::services::state::{
+    GetAnchorRequest, GetNotesOwnedByRequest, GetOpeningRequest,
+    PreverifyRequest,
+};
+
+use dusk_bytes::{DeserializableSlice, Serializable};
+
+use once_cell::sync::Lazy;
+use rand::prelude::*;
+use rand::rngs::StdRng;
+use rusk::error::Error;
+use rusk::{Result, Rusk};
+
+use microkelvin::{BackendCtor, DiskBackend};
+
+use tracing::info;
+
+use tonic::transport::Server;
+
+use rusk::services::pki::{KeysServer, RuskKeys};
+use rusk::services::prover::{ProverServer, RuskProver};
+use rusk::services::state::StateServer;
+
+use dusk_wallet_core::{
+    self as wallet, Store, Transaction, UnprovenTransaction,
+};
+
+use phoenix_core::{Crossover, Fee, Note};
+
+use dusk_bls12_381::BlsScalar;
+use dusk_jubjub::{JubJubAffine, JubJubScalar};
+use dusk_plonk::proof_system::Proof;
+
+use dusk_poseidon::tree::PoseidonBranch;
+use rusk_abi::POSEIDON_TREE_DEPTH;
+
+pub fn testbackend() -> BackendCtor<DiskBackend> {
+    BackendCtor::new(DiskBackend::ephemeral)
+}
+
+static STATE_LOCK: Lazy<Mutex<Rusk>> = Lazy::new(|| {
+    let state_id = rusk_recovery_tools::state::deploy(&testbackend())
+        .expect("Failed to deploy state");
+
+    let mut rusk = Rusk::builder(testbackend)
+        .id(state_id)
+        .build()
+        .expect("Error creating Rusk Instance");
+
+    generate_note(&mut rusk).expect("Failed to generate note");
+
+    Mutex::new(rusk)
+});
+
+const BLOCK_HEIGHT: u64 = 1;
+
+pub static SSK: Lazy<SecretSpendKey> = Lazy::new(|| {
+    info!("Generating SecretSpendKey");
+    TestStore.retrieve_ssk(0).expect("Should not fail in test")
+});
+
+fn generate_note(rusk: &mut Rusk) -> Result<()> {
+    info!("Generating a note");
+    let mut rng = StdRng::seed_from_u64(0xdead);
+
+    let psk = SSK.public_spend_key();
+
+    let initial_balance = 1_000_000_000_000;
+
+    let note = Note::transparent(&mut rng, &psk, initial_balance);
+
+    let mut rusk_state = rusk.state()?;
+    let mut transfer = rusk_state.transfer_contract()?;
+
+    transfer.push_note(BLOCK_HEIGHT, note)?;
+    transfer.update_root()?;
+
+    info!("Updating the new transfer contract state");
+    unsafe {
+        rusk_state
+            .set_contract_state(&rusk_abi::transfer_contract(), &transfer)?;
```

Comment:
This looks like an unsafe overkill to solve a simple problem

Reply:
The reason of `unsafe` is explained in the "Safety" section of the comment, and was already agreed when the method was landed.